lightning/restore/checksum.go

```diff
@@ -267,18 +272,33 @@ func (e *tikvChecksumManager) checksumDB(ctx context.Context, tableInfo *TidbTab
 		return nil, errors.Trace(err)
 	}
 
-	res, err := executor.Execute(ctx, e.client, func() {})
-	if err != nil {
-		return nil, errors.Trace(err)
+	distSQLScanConcurrency := int(e.distSQLScanConcurrency)
+	for i := 0; i < maxErrorRetryCount; i++ {
+		_ = executor.Each(func(request *kv.Request) error {
+			request.Concurrency = distSQLScanConcurrency
+			return nil
+		})
+		var execRes *tipb.ChecksumResponse
+		execRes, err = executor.Execute(ctx, e.client, func() {})
+		if err == nil {
+			return &RemoteChecksum{
+				Schema:     tableInfo.DB,
+				Table:      tableInfo.Name,
+				Checksum:   execRes.Checksum,
+				TotalBytes: execRes.TotalBytes,
+				TotalKVs:   execRes.TotalKvs,
+			}, nil
+		}
+
+		log.L().Warn("remote checksum failed", zap.String("db", tableInfo.DB),
```

Comment:
could we distinguish timeout error and only retry for that?

Reply:
Add check whether is the error is retryable